test/e2e/main_test.go

```diff
@@ -3,9 +3,77 @@ package e2e
 import (
 	"testing"
 
-	f "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/keycloak/keycloak-operator/pkg/apis"
+	keycloakv1alpha1 "github.com/keycloak/keycloak-operator/pkg/apis/keycloak/v1alpha1"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
 )
 
+type deployedOperatorTestStep func(*testing.T, *framework.Framework, *framework.TestCtx, string) error
+type environmentInitializationStep func(*testing.T, *framework.Framework, *framework.TestCtx, string) error
+
+type CRDTest interface {
+	PrepareEnvironment() []environmentInitializationStep
+
+	GetSubsetOfTests() map[string]deployedOperatorTestStep
+}
+
 func TestMain(m *testing.M) {
```

Comment:
I would like to keep this file as is. This file was bootstrapped by Operator SDK and it contains Operator SDK sanity tests. At some point, it is very likely it will go away (but as far as I know, it's still a good practice to have it).

Reply:
Perhaps testing machinery should be moved to a separate file...